pkg/api/apis/operators/v1alpha1/subscription.go

```diff
@@ -0,0 +1,33 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// SetPhase sets the current phase and adds a condition if necessary
+func (c *Subscription) SetSubscriptionPhase(phase SubscriptionPhase, reason ConditionReason, message string) {
```

Comment:
👍 this matches how we're doing conditions on the other resources.

In the future (for the next apiVersion of these objects) I think we should change these to more align with upstream condition use (i.e. not just a record of phase transitions).